internal/pkg/skuba/addons/dex_test.go

```diff
@@ -0,0 +1,192 @@
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "render dex Image URL when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Dex,
+					templater:         renderDexTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetDexImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/caasp-dex:([[:digit:]]{1,}.){2}[[:digit:]]{1,}(-rev[:digit:]{1,})?",
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.DexImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			if !matched {
+				t.Errorf("renderContext.DexImage() = %v, want %v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func Test_renderContext_GangwayClientSecret(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "get gangway client secret when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Dex,
+					templater:         renderDexTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetDexImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: "[[:alpha:][:digit:]]{20}",
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.GangwayClientSecret()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			if !matched {
+				t.Errorf("GetDexImage() = %v, want %v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func Test_dexCallbacks_beforeApply(t *testing.T) {
+	type test struct {
+		name               string
+		addonConfiguration AddonConfiguration
+		skubaConfiguration *skuba.SkubaConfiguration
+		wantErr            bool
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "test dex callbacks beforeApply when cluster version is " + ver.String(),
+			addonConfiguration: AddonConfiguration{
+				ClusterVersion: ver,
+				ControlPlane:   "",
+				ClusterName:    "",
+			},
+			skubaConfiguration: &skuba.SkubaConfiguration{
+				AddonsVersion: kubernetes.AddonsVersion{},
+			},
+			wantErr: true,
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			d := dexCallbacks{}
+			if err := d.beforeApply(tt.addonConfiguration, tt.skubaConfiguration); (err != nil) != tt.wantErr {
+				t.Errorf("dexCallbacks.beforeApply() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_dexCallbacks_afterApply(t *testing.T) {
```

Comment:
This function return nil right now, what's this test case want to test?https://github.com/SUSE/skuba/blob/master/internal/pkg/skuba/addons/dex.go#L77-L79